gammapy/scripts/image_pipe.py

```diff
@@ -2,56 +2,180 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 import logging
+import numpy as np
+from astropy.coordinates import Angle
+from astropy.units import Quantity
+from astropy.io import fits
 from ..utils.scripts import get_parser
+from ..background import fill_acceptance_image
+from ..image import SkyMap, SkyMapCollection, disk_correlate
+from ..stats import significance
 
 __all__ = ['ImageAnalysis']
 
 log = logging.getLogger(__name__)
 
 
-def image_pipe_main(args=None):
-    parser = get_parser(ImageAnalysis)
-    parser.add_argument('config_file', type=str,
-                        help='Config file in YAML format')
-    # TODO: add option to dump the default config file
-    args = parser.parse_args(args)
-    analysis = ImageAnalysis.from_yaml(args.config_file)
-    analysis.run()
-
-
 class ImageAnalysis(object):
     """Gammapy 2D image based analysis.
+
+    Parameters
+    ----------
+    empty_image : `~gammapy.image.SkyMap`
+            ref to an empty image
+    energy_band : `~gammapy.utils.energy.Energy
+        Energy band for which we want to compute the image
+    offset_band :`astropy.coordinates.Angle`
+        Offset Band where you compute the image
+    data_store : `~gammapy.data.DataStore`
+        `DataStore` where are situated the events
+    counts_image : ~gammapy.image.SkyMap`
+            count image
+    bkg_image : ~gammapy.image.SkyMap`
+            bkg image
     """
 
-    def __init__(self, config):
-        self.config = config
-
-    @classmethod
-    def from_yaml(cls, filename):
-        """Read config from YAML file."""
-        import yaml
-        log.info('Reading {}'.format(filename))
-        with open(filename) as fh:
-            config = yaml.safe_load(fh)
-        return cls(config)
-
-    def run(self):
-        """Run analysis chain."""
-        log.info('Running analysis ...')
-        print(self.config['general']['outdir'])
-        print(self.config)
-
-    def make_counts(self):
+    def __init__(self, empty_image=None,
+                 energy_band=None, offset_band=None,
+                 data_store=None, counts_image=None, bkg_image=None):
+        self.maps = SkyMapCollection()
+        if counts_image:
+            self.maps["total_counts"] = counts_image
+        else:
+            self.data_store = data_store
+            self.energy_band = energy_band
+            self.offset_band = offset_band
+
+            self.empty_image = empty_image
+            self.header = self.empty_image.to_image_hdu().header
+        if bkg_image:
+            self.maps["total_bkg"] = bkg_image
+
+    def make_counts(self, obs_id):
+        """Fill the counts image for the events of one observation.
+
+        Parameters
+        ----------
+        obs_id : int
+            Number of the observation
+
+        """
+        self.maps["counts"] = SkyMap.empty_like(self.empty_image)
```

Comment:
Just wanted to mention that the `SkyMapCollection` class allows attribute access as well. i.e. `self.maps.counts` instead of `self.maps['counts']`, but I don't have a preference what to use...